amp/WEB-INF/src/org/dgfoundation/amp/ar/AmpARFilter.java

```diff
@@ -878,19 +880,19 @@ public void fillWithDefaultsFilter(Long ampReportId)
         this.initFilterQuery();
 
         getEffectiveSettings(); // do not remove call - also writes into the caches
-        
-        TeamMember tm = null;
-        
-        if (request != null)
-            tm = (TeamMember) request.getSession().getAttribute(Constants.CURRENT_MEMBER);
-        
-        if (tm == null || tm.getTeamId() == null )
-            tm  = null;
 
-        if (tm != null) {
+        if (teamMember == null && request != null) {
+            teamMember = (TeamMember) request.getSession().getAttribute(Constants.CURRENT_MEMBER);
+        }
```

Comment:
I am wondering if `teamMember` wasn't stored within `AmpARFilter` on purpose. E.g. if user logs out and refreshes a report, will it work ok? Or could there be any other "cached" filter issue?

Reply:
AmpARFilter object is stored in the user session. If the user logs out, the object will not be available anymore.

Reply:
will it update correctly when switching workspace?

Reply:
Yes, you are right. It is recreated every time when the user switches to a  new workspace